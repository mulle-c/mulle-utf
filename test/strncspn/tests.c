#include <mulle-utf/mulle-utf.h>

#include <stdio.h>


static void   test( char *s, size_t len, char *p)
{
   size_t  result;

   printf( "mulle_utf8_strncspn( ");
   if( ! s)
      printf( "NULL");
   else
      printf( "\"%.*s\"", (int) len, s);

   printf( ", %d", (int) len);

   if( ! p)
      printf( ", NULL");
   else
      printf( ", \"%s\"", p);

   printf( ") = ");

   result = mulle_utf8_strncspn( (char *) s, len, (char *) p);
   printf( "%ld (%ld)\n", (long) result, (long) ((s && p) ? strcspn( s, p) : 0));
}



int   main( int argc, char **argv)
{
   test( NULL, 0, NULL);
   test( "", 0, NULL);
   test( NULL, 0, "");
   test( "", 0, "");
   test( "", 1, "");

   test( NULL, 2, "");

   test( "abcd", -1, "a");
   test( "abcd", 4, "b");
   test( "abcd", 3, "d");
   test( "abcd", 4, "d");
   test( "abcd", 4, "xyz");

   test( "abcd", -1, "xdy");

   test( "abcd", 4, "ab");
   test( "abcd", 4, "bc");

   test( "abcd", 4, "abc");

   test( "abc", 3, "b");
   test( "abc", 3, "abc");

   return( 0);
}