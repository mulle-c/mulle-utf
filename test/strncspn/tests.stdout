mulle_utf8_strncspn( NULL, 0, NULL) = 0 (0)
mulle_utf8_strncspn( "", 0, NULL) = 0 (0)
mulle_utf8_strncspn( NULL, 0, "") = 0 (0)
mulle_utf8_strncspn( "", 0, "") = 0 (0)
mulle_utf8_strncspn( "", 1, "") = 1 (0)
mulle_utf8_strncspn( NULL, 2, "") = 0 (0)
mulle_utf8_strncspn( "abcd", -1, "a") = 0 (0)
mulle_utf8_strncspn( "abcd", 4, "b") = 1 (1)
mulle_utf8_strncspn( "abc", 3, "d") = 3 (3)
mulle_utf8_strncspn( "abcd", 4, "d") = 3 (3)
mulle_utf8_strncspn( "abcd", 4, "xyz") = 4 (4)
mulle_utf8_strncspn( "abcd", -1, "xdy") = 3 (3)
mulle_utf8_strncspn( "abcd", 4, "ab") = 0 (0)
mulle_utf8_strncspn( "abcd", 4, "bc") = 1 (1)
mulle_utf8_strncspn( "abcd", 4, "abc") = 0 (0)
mulle_utf8_strncspn( "abc", 3, "b") = 1 (1)
mulle_utf8_strncspn( "abc", 3, "abc") = 0 (0)
